app.js

```diff
@@ -0,0 +1,41 @@
+import express from 'express';
+import cors from 'cors';
+const app = express()
+app.use(express.json())
+app.use(cors());
+
+app.locals.notes = [];
+
+app.get('/notes', (req, res) => {
+  res.status(200).json(app.locals.notes);
+});
+
+// We are not sure at this point if we want to use this? 
+// app.get('/notes/:id', (req, res) => {
+//   const note = app.locals.notes.find((note) => note.id == req.params.id);
+//   if (!note) return res.status(404).json('Note not found.');
+//   res.status(200).json(note);
+// })
+
+app.delete('/notes/:id', (req, res) => {
+  const noteIndex = app.locals.notes.findIndex(note => note.id == req.params.id);
+  if (noteIndex === -1) return res.status(404).json('Note not found');
+  app.locals.notes.splice(noteIndex, 1);
+  return res.status(204).send('Note deleted');
+})
+
+app.post('/notes', (req, res) => {
+  const { id, title, noteItems } = req.body;
+  const isduplicate = app.locals.notes.some(note => note.id == id)
+  if (!id) return res.status(422).json('Error posting note');
+  if (isduplicate) return res.status(409).json('Please try adding your note again')
```

Comment:
🔥 Great check here for duplicates!!

Reply:
Also,  I understand that it is more convenient during this stage in development for the user to resubmit than to import uuid. I'll open an issue regarding the usage of uuid in the backend!

I imagine we could use it here as a duplicate would not be the fault of the user, it would be the random chance uuid gave the same id lol. So we could generate a new one in the backend, or have the user resubmit which would create a new id in the front-end as well.

We could also check in the front-end if there is a note with the id before submitting to the backend.

So many paths!!!!!

Reply:
I also agree on the check for duplicates. Thanks for opening an issue regarding this.